### PR TITLE
Feature/tag autocomplete

### PR DESCRIPTION
**Pull request by samshara**

### CODE REVIEW

File: src/views/Home/Filters/TagMultiSelectInput/index.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        pending: tagSearchPending,
+        response: tagSearchResponse,
+    } = useRequest<MultiResponse<string>>({
+        url: '/api/autocomplete/tags',
+        method: 'GET',
+        skip: !opened || isFalsyString(debouncedSearchText),
+        query: searchQueryParams,
+    });
+
+    const tags = tagSearchResponse?.results.map((tag) => ({ id: tag, label: tag }));
```
Comment (AdityaKhatri) [inline src/views/Home/Filters/TagMultiSelectInput/index.tsx:46]:
```suggestion
    const tags = tagSearchResponse?.results?.map((tag) => ({ id: tag, label: tag }));
```

File: src/views/Home/Filters/TagMultiSelectInput/index.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    const searchQueryParams = useMemo(() => ({
+        keyword: debouncedSearchText,
+    }), [debouncedSearchText]);
+
+    const {
+        pending: tagSearchPending,
+        response: tagSearchResponse,
+    } = useRequest<MultiResponse<string>>({
+        url: '/api/autocomplete/tags',
```
Comment (AdityaKhatri) [inline src/views/Home/Filters/TagMultiSelectInput/index.tsx:40]:
```suggestion
        url: '/api/autocomplete/tags/',
```

File: src/views/Home/Filters/TagMultiSelectInput/index.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+const labelSelector = (d: Tag) => d.label;
+
+function TagMultiSelectInput<K extends string>(props: TagSelectInputProps<K>) {
+    const {
+        className,
+        ...otherProps
+    } = props;
+
+    const [opened, setOpened] = useState(false);
+    const [searchText, setSearchText] = useState<string>('');
```
Comment (AdityaKhatri) [inline src/views/Home/Filters/TagMultiSelectInput/index.tsx:29]:
```suggestion
    const [searchText, setSearchText] = useState('');
```

File: src/views/Home/Filters/TagMultiSelectInput/index.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    string,
+    K,
+    Tag,
+    Def,
+    'onSearchValueChange' | 'searchOptions' | 'optionsPending' | 'keySelector' | 'labelSelector' | 'totalOptionsCount' | 'onShowDropdownChange'
+>;
+const keySelector = (d: Tag) => d.id;
+const labelSelector = (d: Tag) => d.label;
+
+function TagMultiSelectInput<K extends string>(props: TagSelectInputProps<K>) {
```
Comment (AdityaKhatri) [inline src/views/Home/Filters/TagMultiSelectInput/index.tsx:22]: Have type next to the function

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]
Fix review requests

File: src/views/Home/Filters/TagMultiSelectInput/index.tsx (hunk 2 above)
Comment (samshara) [reply src/views/Home/Filters/TagMultiSelectInput/index.tsx:40]: @AdityaKhatri the API doesn't work with the / attached

Comment (AdityaKhatri) [reply src/views/Home/Filters/TagMultiSelectInput/index.tsx:41]: oh no

**Pull request merged**